### PR TITLE
Remove EnumSet litter from MetricsCollectionCycle

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Implements enhancement described here: https://github.com/hazelcast/hazelcast/pull/15667#discussion_r331541195

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class MetricTargetTest {
+
+    @Test
+    public void testAsSet_returnsSameObjects() {
+        assertTrue("set objects must be cached and reused",
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java]: We can use `assertSame` here.

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                asSet(new MetricTarget[]{MANAGEMENT_CENTER}) == asSet(new MetricTarget[]{MANAGEMENT_CENTER}));
+    }
+
+    @Test
+    public void testAsSet_ignoresPermutations() {
+        assertEquals(
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java]: Here we should use `assertSame` I think.

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        assertAsSetContainsAll(JMX);
+        assertAsSetContainsAll(DIAGNOSTICS);
+        assertAsSetContainsAll(DIAGNOSTICS, JMX);
+        assertAsSetContainsAll(DIAGNOSTICS, MANAGEMENT_CENTER);
+        assertAsSetContainsAll(MANAGEMENT_CENTER, JMX);
+        assertAsSetContainsAll(MANAGEMENT_CENTER, JMX, DIAGNOSTICS);
+    }
+
+    private void assertAsSetContainsAll(MetricTarget... targets) {
+        assertContainsAll(asSet(targets), asList(targets));
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java]: Minor: I wouldn't static import these for clarity.

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java (hunk 2 above)
Comment (puzpuzpuz) [reply hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java]: I've overlooked `assertSame`. Thanks for the suggestion! Addressed.

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java (hunk 1 above)
Comment (puzpuzpuz) [reply hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java]: Addressed

File: hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java (hunk 3 above)
Comment (puzpuzpuz) [reply hazelcast/src/test/java/com/hazelcast/internal/metrics/MetricTargetTest.java]: Makes sense. Addressed

**Comment by puzpuzpuz** [comment]
I've pushed a minor improvement into this branch: now `MetricTarget#asSet` can handle empty array correctly by returning `emptySet()`. This way it looks more consistent. It also remove one of methods from `MetricsCollectionCycle`.

**Pull request merged**